test/meetup_cache_test.exs

```diff
@@ -0,0 +1,804 @@
+        event_url: "http://old2.com",
+        datetime: tomorrow_date,
+        end_datetime: tomorrow_date
+      }
+      |> Repo.insert!()
+
+      # Replaces with completely new set
+      events = [
+        %{
+          source: "manual",
+          source_id: "manual-new-1",
+          name: "New Manual 1",
+          title: "Brand new event",
+          event_url: "http://new1.com",
+          datetime: day_after_tomorrow_date,
+          end_datetime: day_after_tomorrow_date
+        },
+        %{
+          source: "manual",
+          source_id: "manual-new-2",
+          name: "New Manual 2",
+          title: "Another new event",
+          event_url: "http://new2.com",
+          datetime: next_week_date,
+          end_datetime: next_week_date
+        }
+      ]
+
+      MeetupCache.sync_manual_events(events)
+
+      stored_events = MeetupCache.all()
+      assert length(stored_events) == 2
+
+      source_ids = Enum.map(stored_events, & &1.source_id)
+      assert "manual-new-1" in source_ids
+      assert "manual-new-2" in source_ids
+      refute "manual-old-1" in source_ids
+      refute "manual-old-2" in source_ids
+    end
+
+    test "handles mix of creating new and updating existing manual events" do
+      yesterday_date = yesterday()
+      tomorrow_date = tomorrow()
+      next_week_date = next_week()
+
+      # Existing manual event that will be updated
+      %Event{
+        source: "manual",
+        source_id: "manual-update",
+        name: "Old Manual Name",
+        title: "Will be updated",
+        event_url: "http://old-manual.com",
+        datetime: yesterday_date,
+        end_datetime: yesterday_date
+      }
+      |> Repo.insert!()
+
+      # Mix of update and new events
+      events = [
+        %{
+          source: "manual",
+          source_id: "manual-update",
+          name: "Updated Manual Name",
+          title: "Was updated",
+          event_url: "http://updated-manual.com",
+          datetime: tomorrow_date,
+          end_datetime: tomorrow_date
+        },
+        %{
+          source: "manual",
+          source_id: "manual-brand-new",
+          name: "Brand New Manual",
+          title: "Completely new",
+          event_url: "http://brand-new.com",
+          datetime: next_week_date,
+          end_datetime: next_week_date
+        }
+      ]
+
+      MeetupCache.sync_manual_events(events)
+
+      stored_events = MeetupCache.all()
+      assert length(stored_events) == 2
+
+      updated_event = Enum.find(stored_events, &(&1.source_id == "manual-update"))
+      assert updated_event.name == "Updated Manual Name"
+      assert updated_event.title == "Was updated"
+      assert updated_event.datetime == tomorrow_date
+
+      new_event = Enum.find(stored_events, &(&1.source_id == "manual-brand-new"))
+      assert new_event.name == "Brand New Manual"
+      assert new_event.title == "Completely new"
+      assert new_event.datetime == next_week_date
+    end
+  end
+end
```

Comment:
Do you think all these tests are needed, worth it? 🤔

Reply:
I wanted to have a very robust test suite to encourage more devs to feel more confident contributing, since manually reproducing all these different cases by hand is quite time consuming. But we can delete some for sure. Which one would you omit?